bambou/error.go

```diff
@@ -27,66 +27,41 @@ import (
 	"fmt"
 )
 
-const (
-	// ErrorCodeJSONCannotDecode is the code that means it is impossible to
-	// decode json data.
-	ErrorCodeJSONCannotDecode = 10001
-
-	// ErrorCodeJSONCannotEncode is the code that means it is impossible to
-	// encode json data.
-	ErrorCodeJSONCannotEncode = 10002
-
-	// ErrorCodeSessionAlreadyStarted is the error code that means a session
-	// is already stared.
-	ErrorCodeSessionAlreadyStarted = 11001
-
-	// ErrorCodeSessionCannotForgetAuthToken is the code that means no password
-	// or token has been given to the session.
-	ErrorCodeSessionCannotForgeAuthToken = 11002
-
-	// ErrorCodeSessionCannotProcessRequest is the code that means that it was
-	// impossible to process a request.
-	ErrorCodeSessionCannotProcessRequest = 11003
-
-	// ErrorCodeSessionIDNotSet is the code that means the Identifiable is
-	// missing a required ID.
-	ErrorCodeSessionIDNotSet = 11004
+type VsdErrorList struct {
+	VsdErrors    []VsdError `json:"errors"`
+	VsdErrorCode int        `json:"internalErrorCode"`
+}
 
-	// ErrorCodeSessionUsernameNotSet is the code that means the username
-	// is missing.
-	ErrorCodeSessionUsernameNotSet = 11005
-)
+type VsdError struct {
+	Property     string  `json:"property"`
+	Descriptions []Error `json:"descriptions"` // XXX -- note
```

Comment:
Not really sure if it is good to add comments like `// XXX -- note`, either we need a good comment on VsdError explaining what it is (and why you want to put a 'note', without putting it), or we should not mention it. At least, comments behind code is not generally nice in my opinion